zebra/zebra_rib.c

```diff
@@ -2224,6 +2226,8 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		if (re->type == ZEBRA_ROUTE_BGP
 		    && CHECK_FLAG(re->flags, ZEBRA_FLAG_IBGP))
 			re->distance = 200;
+		else if (re->type == ZEBRA_ROUTE_KERNEL)
+			re->distance = re->metric;
```

Comment:
I am also pretty sure we should not consider the metric as inserted into the kernel from an outside source as the actual admin distance.  This change scares me.